File: benchmarks/cx-vs-python/python-benchmarks/ackermann-function.py
import time


def ack1(M, N):
    return (N + 1) if M == 0 else (
        ack1(M-1, 1) if N == 0 else ack1(M-1, ack1(M, N-1)))


def test():
    start = time.time()
    print "(0, 5)"
    print "Result: {value}".format(value=ack1(0, 5))

    end = time.time()
    print (end - start)


test()

File: benchmarks/golang-benchmarks/ackermann/ackermann-function.go
package ackermann

import (
	"fmt"
)

func ackermann(m, n uint) uint {
	switch uint(0) {
	case m:
		return n + uint(1)
	case n:
		return ackermann(m-uint(1), uint(1))
	}
	return ackermann(n-uint(1), ackermann(m, n-uint(1)))
}

func main() {
	fmt.Printf("Ackermann Function")
	fmt.Printf("Test with (3, 1)")
	fmt.Printf("Result: ")
	fmt.Println(ackermann(uint(3), uint(1)))
}
